### PR TITLE
[WIP] Bad identifier suggestion in certain cases

### LINKED ISSUE #4156: Bad identifier suggestion in certain cases
In the following:

```
contract c {
    function c ()
    {
        a = foo.bar;
    }
    uint256 a;
}
```

instead of `foo` the compiler suggests `now`:

```
contract c {
^ (Relevant source part starts here and spans across multiple lines).
bad_suggestion.sol:4:13: Error: Undeclared identifier. Did you mean "now"?
        a = foo.bar;
            ^-^
```

(This actually is in a test: `SolidityParser/enum_valid_declaration`)

### PR DESCRIPTION
**Pull request by mestorlx**
Fixes #4156
I Updated the MAXIMUM_STRING_DISTANCE based on the length of the identifier that is being searched as follow:
1: search(no changes)
2-3: at most one change
4-: at most two changes

@chriseth I think this would be enough for the first part. Any comments?

### CODE REVIEW

File: libsolidity/analysis/DeclarationContainer.cpp [hunk 1]
```diff
@@ -137,13 +137,25 @@ vector<Declaration const*> DeclarationContainer::resolveName(ASTString const& _n
...
 {
-	static size_t const MAXIMUM_EDIT_DISTANCE = 2;
+
 	// because the function below has quadratic runtime - it will not magically improve once a better algorithm is discovered ;)
 	// since 80 is the suggested line length limit, we use 80^2 as length threshold
 	static size_t const MAXIMUM_LENGTH_THRESHOLD = 80 * 80;
 
 	vector<ASTString> similar;
-
+	static size_t MAXIMUM_EDIT_DISTANCE = 0;
```
Comment (chriseth) [inline libsolidity/analysis/DeclarationContainer.cpp]: This should not be `static` anymore. Also, it should use camelCase now. Finally, it is probably better to use something like `maximumEditDistance = _name.size() > 3 ? 2 : _name.size() / 2;`

Comment (mestorlx) [reply libsolidity/analysis/DeclarationContainer.cpp]: Thats smart!
Updated

**Review by chriseth — CHANGES_REQUESTED** [review]
Great! Please also add an entry into Changelog.md in the "features" section.

**Comment by mestorlx** [comment]
Added.

**Review by chriseth — APPROVED** [review]

**Pull request merged**